src/services/AppService.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ *  AppService.ts
+ *
+ *  For application wide services
+ *  @copyright 2024 Digital Aid Seattle
+ *
+ */
+
+import supabaseClient from "../../utils/supabaseClient";
+
+
+type AppConstant = {
+  value: string,
+  label: string
+}
+
+class AppService {
+
+  cache: Map<string, AppConstant[]> = null;
+
+  // constructor() {
+
+  // We could do this instead of lazy-init in the getAppConstants, 
+  // but testing gets messed up.
+
+  // this.lookup()
+  //   .then(m => this.cache = m)
+  //   .catch(err => console.error(err))
+  // }
+
+  getAppConstants = async (type: string): Promise<AppConstant[]> => {
+    if (this.cache === null) {
+      return this.lookup()
+        .then(m => {
+          this.cache = m;
+          return this.cache.get(type);
+        })
+    } else {
+      return Promise.resolve(this.cache.get(type));
```

Comment:
Don't need to wrap the value in a Promise as the function is marked async which wraps the value automatically

Reply:
thx